src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+
+        if ((fabsf(player->actor.world.pos.y - sp30.y) < 50.0f) && (distXZ < arg2)) {
+            arg1->unk_01++;
+        }
+
+        return 0;
+    }
+
+    return 1;
+}
+
+void func_80122868(GlobalContext* globalCtx, Player* player) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (player->invincibilityTimer > 0) {
+        s32 phi_v0 = 50 - player->invincibilityTimer;
+
+        phi_v0 = CLAMP(phi_v0, 8, 40);
+
+        player->unk_B5F += phi_v0;
+        POLY_OPA_DISP =
+            Gfx_SetFog(POLY_OPA_DISP, 255, 0, 0, 0, 0, 4000 - (s32)(Math_CosS((player->unk_B5F << 8)) * 2000.0f));
+    } else if (gSaveContext.unk_1016 != 0) {
+        player->unk_B5F += 10;
+        POLY_OPA_DISP =
+            Gfx_SetFog(POLY_OPA_DISP, 0, 0, 255, 0, 0, 4000 - (s32)(Math_CosS((player->unk_B5F << 8)) * 2000.0f));
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_801229A0(GlobalContext* globalCtx, Player* player) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if ((gSaveContext.unk_1016 != 0) || (player->invincibilityTimer > 0)) {
+        POLY_OPA_DISP = func_801660B8(globalCtx, POLY_OPA_DISP);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_801229EC(UNK_TYPE arg0, UNK_TYPE arg1) {
+}
+
+extern s16 D_801BFDA0[PLAYER_MASK_MAX - 1];
+#if 0
+s16 D_801BFDA0[PLAYER_MASK_MAX - 1] = {
+    OBJECT_MASK_TRUTH,  OBJECT_MASK_KERFAY,  OBJECT_MASK_YOFUKASI, OBJECT_MASK_RABIT,   OBJECT_MASK_KI_TAN,
+    OBJECT_MASK_JSON,   OBJECT_MASK_ROMERNY, OBJECT_MASK_ZACHO,    OBJECT_MASK_POSTHAT, OBJECT_MASK_MEOTO,
+    OBJECT_MASK_BIGELF, OBJECT_MASK_GIBUDO,  OBJECT_MASK_GERO,     OBJECT_MASK_DANCER,  OBJECT_MASK_SKJ,
+    OBJECT_MASK_STONE,  OBJECT_MASK_BREE,    OBJECT_MASK_BAKURETU, OBJECT_MASK_BU_SAN,  OBJECT_MASK_KYOJIN,
+    OBJECT_MASK_BOY,    OBJECT_MASK_GORON,   OBJECT_MASK_ZORA,     OBJECT_MASK_NUTS,
+};
+#endif
+
+// Load mask?
+void func_801229FC(Player* player) {
+    if (player->maskObjectLoading == 1) {
+        // TODO: check if player->maskId is unsigned
+        s16 objectId = D_801BFDA0[(u8)player->maskId - 1];
+
+        osCreateMesgQueue(&player->maskObjectLoadQueue, &player->maskObjectLoadMsg, 1);
+        DmaMgr_SendRequestImpl(&player->maskDmaRequest, player->maskObjectSegment, gObjectTable[objectId].vromStart,
+                               gObjectTable[objectId].vromEnd - gObjectTable[objectId].vromStart, 0,
+                               &player->maskObjectLoadQueue, NULL);
+        player->maskObjectLoading++;
+    } else if (player->maskObjectLoading == 2) {
+        if (osRecvMesg(&player->maskObjectLoadQueue, NULL, OS_MESG_NOBLOCK) == 0) {
+            player->maskObjectLoading = 0;
+
+            if (player->currentMask == PLAYER_MASK_GREAT_FAIRY) {
+                s32 i;
+
+                for (i = 0; i < ARRAY_COUNT(D_801F58B0); i++) {
+                    func_80127B64(D_801F58B0[i], ARRAY_COUNT(D_801F58B0[i]), &player->bodyPartsPos[7]);
```

Comment:
Presumably this is some sort of extra function for the animated stuff on Great Fairy's Mask. Is the plan to not name stuff like this until later player_lib PRs?

Reply:
This is the function.
https://github.com/Decompollaborate/mm/blob/3f7ff526991af1b5e67b0457ca9f8c8e35808253/src/code/z_player_lib.c#L2809-L2818
It just performs some Vec3f copies.

The only other place where `D_801F58B0` is used is in `func_801284A0`, which is currently non equivalent
https://github.com/Decompollaborate/mm/blob/3f7ff526991af1b5e67b0457ca9f8c8e35808253/src/code/z_player_lib.c#L3037-L3069